src/signalrclient/case_insensitive_comparison_utils.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <functional>
+#include <cctype>
+#include "cpprest\details\basic_types.h"
+
+namespace signalr
+{
+    // Note: These functions are not pretending to be all-purpose helpers for case insensitive string comparison. Rather
+    // we use them to compare hub and hub method names which are expected to be almost exclusively ASCII and this is the
+    // simplest thing that would work without having to take dependencies on third party libraries.
+    struct case_insensitive_equals : std::binary_function<utility::string_t, utility::string_t, bool>
+    {
+        bool operator()(const utility::string_t& s1, const utility::string_t& s2) const
+        {
+            if (s1.length() != s2.length())
+            {
+                return false;
+            }
+
+            for (auto s1_iterator = s1.begin(), s2_iterator = s2.begin(); s1_iterator != s1.end() && s2_iterator != s2.end(); ++s1_iterator, ++s2_iterator)

```

Comment:
nit: Technically don't need the second check in the for loop since you already checked that they are the same length